hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/impl/BaseTransactionalDataRegion.java

```diff
@@ -148,8 +144,20 @@ protected void prepareForValidation() {
 			assert strategy == Strategy.VALIDATION;
 			return;
 		}
-		validator = new PutFromLoadValidator(cache, factory);
-		strategy = Strategy.VALIDATION;
+		// If two regions share the same name, they should use the same cache.
```

Comment:
@rvansa does this BaseTransactionalDataRegion change look correct for ORM 5.1?  I'm asking in case BaseTransactionalDataRegion is dependent on other code that might be different between ORM 5.1 + 5.2 (not that I know of any differences).  Thanks!  -Scott

Reply:
Not sure why are you concerned... The concept of the fix should apply to both branches imo.